src/main/java/de/zalando/aruha/nakadi/validation/EventValidation.java

```diff
@@ -51,10 +49,33 @@ private static JSONObject wrapSchemaInData(final JSONObject schema) {
         return wrapper;
     }
 
-    private static JSONObject addMetadata(final JSONObject schema) {
+    private static JSONObject addMetadata(final JSONObject schema, EventType eventType) {
         normalizeSchema(schema);
 
-        schema.getJSONObject("properties").put("metadata", new JSONObject("{\"type\": \"object\"}"));
+        final JSONObject metadata = new JSONObject();
+        final JSONObject metadataProperties = new JSONObject();
+
+        final JSONObject uuid = new JSONObject("{\"type\": \"string\", \"pattern\": \"^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$\"}");
+        final JSONObject eventTypeString = new JSONObject("{\"type\": \"string\", \"enum\": [\"" + eventType.getName() + "\"]}");
+        final JSONObject string = new JSONObject("{\"type\": \"string\"}");
+
+        // We know this is not rfc3339. We think this is enough by now
+        final String rfc3339 = "^[0-9]{4}-?[0-9]{2}-?[0-9]{2}T?[0-9]{2}:?[0-9]{2}:?[0-9]{2}(.[0-9]+)?(Z|[+-][0-9]{2}:?[0-9]{2})$";
```

Comment:
I think it would be really great if in our API for `occurred_at` field we specify an example so that our users will know which date-format they can specify.